### PR TITLE
Safely open config

### LINKED ISSUE #227: Allow for some values to not be defined in config
Code will panic if config exists and only one of two values are defined

### PR DESCRIPTION
**Pull request by JakeRoggenbuck**
fixes #227

### CODE REVIEW

**Comment by JakeRoggenbuck** [comment]
tested

File: src/config.rs [hunk 1]
```diff
@@ -22,24 +22,67 @@ pub fn config_dir_exists() -> bool {
...
+    pub overheat_threshold: Option<i8>,
+}
+
+trait SafeFillConfig {
+    fn safe_fill_config(&mut self) -> Config;
+}
+
+impl SafeFillConfig for SafeConfig {
+    fn safe_fill_config(&mut self) -> Config {
+        // This function makes sure the config contains ever value from Config,
```
Comment (adamhutchings) [inline src/config.rs:50]: every?

**Review by Shuzhengz — APPROVED** [review]
tested (but not under sudo yet), lgtm

**Comment by JakeRoggenbuck** [comment]
Yee, i can double check for sudo, thanks!

**Comment by JakeRoggenbuck** [comment]
Works with sudo

**Pull request merged**